Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,533 @@
+        var e = defaultValue(ellipsoid, Ellipsoid.WGS84);
+        this._ellipsoid = e;
+        this._start = new Cartographic();
+        this._end = new Cartographic();
+
+        this._heading = undefined;
+        this._distance = undefined;
+        this._ellipticity = undefined;
+        this._ellipticitySquared = undefined;
+
+        if (defined(start) && defined(end)) {
+            computeProperties(this, start, end, e);
+        }
+    }
+
+    defineProperties(EllipsoidRhumbLine.prototype, {
+        /**
+         * Gets the ellipsoid.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Ellipsoid}
+         * @readonly
+         */
+        ellipsoid : {
+            get : function() {
+                return this._ellipsoid;
+            }
+        },
+
+        /**
+         * Gets the surface distance between the start and end point
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Number}
+         * @readonly
+         */
+        surfaceDistance : {
+            get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                Check.defined('distance', this._distance);
+                //>>includeEnd('debug');
+
+                return this._distance;
+            }
+        },
+
+        /**
+         * Gets the initial planetodetic point on the path.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Cartographic}
+         * @readonly
+         */
+        start : {
+            get : function() {
+                return this._start;
+            }
+        },
+
+        /**
+         * Gets the final planetodetic point on the path.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Cartographic}
+         * @readonly
+         */
+        end : {
+            get : function() {
+                return this._end;
+            }
+        },
+
+        /**
+         * Gets the heading from the start point to the end point.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Number}
+         * @readonly
+         */
+        heading : {
+            get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                Check.defined('distance', this._distance);
+                //>>includeEnd('debug');
+
+                return this._heading;
+            }
+        }
+    });
+
+    /**
+     * Create a rhumb line using an initial and final position.
+     *
+     * @param {Cartographic} start The initial planetodetic point on the path.
+     * @param {Cartographic} end The final planetodetic point on the path.
+     * @param {Ellipsoid} [ellipsoid=Ellipsoid.WGS84] The ellipsoid on which the rhumb line lies.
+     * @param {EllipsoidRhumbLine} [result] The object in which to store the result.
+     * @returns {EllipsoidRhumbLine} The EllipsoidRhumbLine object.
+     */
+    EllipsoidRhumbLine.fromStartAndEnd = function(start, end, ellipsoid, result) {
```

Comment:
Why is this function needed, it just matches the constructor, doesn't it??

Reply:
I added it because it allows using a scratch result object.

Reply:
Seems like overkill, can't you just call `setEndPoints` on an existing object?  I doubt someone would need to change the ellipsoid on an existing instance.  If we have a valid use case, then maybe it's worth it, but we try not to add code "just because it might be useful" since that introduces more code to maintain.